File: GoLang/solution/solution1161.go
package solution

import (
	datastructure "gproject/dataStructure"
	"math"
)

func MaxLevelSum(root *datastructure.TreeNode) int {
	ans := 1
	max := math.MinInt64

	queue := make([]*datastructure.TreeNode, 0)
	queue = append(queue, root)
	level := 1
	for len(queue) != 0 {
		sum := 0
		tq := make([]*datastructure.TreeNode, 0)
		for _, t := range queue {
			sum += t.Val
			if t.Left != nil {
				tq = append(tq, t.Left)
			}
			if t.Right != nil {
				tq = append(tq, t.Right)
			}
		}
		if sum > max {
			max = sum
			ans = level
		}
		queue = tq
		level++
	}
	return ans
}

File: java/src/Solution1161.java
import dataStructure.TreeNode;

import java.util.LinkedList;
import java.util.Queue;

public class Solution1161 {
	public int maxLevelSum(TreeNode root) {
		int ans = 1, max = Integer.MIN_VALUE;
		Queue<TreeNode> queue = new LinkedList<>();
		queue.add(root);
		int level = 1;
		while (!queue.isEmpty()) {
			int sum = 0;
			Queue<TreeNode> queue1 = new LinkedList<>();
			while (!queue.isEmpty()) {
				TreeNode t = queue.poll();
				sum += t.val;
				if (t.left != null) {
					queue1.add(t.left);
				}
				if (t.right != null) {
					queue1.add(t.right);
				}
			}
			if (sum > max) {
				max = sum;
				ans = level;
			}
			queue = queue1;
			level++;
		}
		return ans;
	}
}
